src/js/modules/setup/controllers/wallet-backup/wallet-backup.ctrl.js

```diff
@@ -4,210 +4,108 @@
-        var backupSettings = {
-            // NB: on android fileOpener2 only works with SD storage (i.e. non-private storage)
-            path: $window.cordova ? ($window.ionic.Platform.isAndroid() ? $window.cordova.file.externalDataDirectory : $window.cordova.file.documentsDirectory) : null,
-            filename: "btc-wallet-backup-" + backupInfo.identifier + ".pdf",
-            replace: true
-        };
+    function SetupWalletBackupCtrl($scope, $q, $state, modalService, launchService, settingsService, walletBackupService) {
+        var walletBackupData;
+        var readOnlySettingsData;
+
+        var actionButtonOptions = [
+            {
+                icon: "ion-email",
+                value: "email"
+            },
+            {
+                icon: "ion-upload",
+                value: "open"
+            }
+        ];
 
-        $scope.isSaveButtonClicked = false;
+        $scope.isButtonEmailOrOpenClicked = false;
         $scope.form = {
             isBackupSaved: false
         };
 
-        // Methods
-        $scope.onSkipBackup = onSkipBackup;
         $scope.onShowExportOptions = onShowExportOptions;
-        $scope.clearBackupInfoAndContinue = clearBackupInfoAndContinue;
+        $scope.onSkipBackup = onSkipBackup;
+        $scope.onNextStep = onNextStep;
 
-        /**
-         * On show export options
-         */
-        function onShowExportOptions() {
-            actionSheet = $ionicActionSheet.show({
-                titleText: "",
-                buttons: [
-                    {text: $translate.instant("BACKUP_EMAIL_PDF")},
-                    {text: $translate.instant("BACKUP_OPEN_PDF")}
-                ],
-                cancelText: "",
-                buttonClicked: onActionSheetButtonClickHandler
-            });
-        }
+        init();
 
         /**
-         * Click handler for action sheet buttons
-         * @param index { integer }
-         * @return { boolean }
+         * Initialize
          */
-        function onActionSheetButtonClickHandler(index) {
-            modalService.showSpinner({
-                title: "",
-                body: "GENERATE_PDF_FILE"
-            });
-
-            switch (index) {
-                case 0:
-                    generatePdf()
-                        .then(emailBackupPdf)
-                        .catch(backupPdfErrorHandler);
-                    break;
-                case 1:
-                    generatePdf()
-                        .then(openBackupPdf)
-                        .catch(backupPdfErrorHandler);
-                    break;
-            }
-
-            return true;
+        function init() {
+            modalService.showSpinner();
+
+            $q.all([launchService.getWalletBackup(), settingsService.getSettings()])
+                .then(function(data) {
+                    walletBackupData = data[0];
+                    readOnlySettingsData = settingsService.getReadOnlySettingsData();
```

Comment:
`settingsService.getSettings()` is very confusing here, I think it should have been named `initSettings` maybe? /discuss